### ISSUE TITLE
Masking layer slows down fitting the model

### ISSUE THREAD

**Issue by CSharpDummy** [keras-team/keras#14452]
When I add a Masking layer to a GRU type network, the model fit becomes very slow by a factor of ~20. Any idea why that might be?

model.add(tf.keras.layers.Masking(mask_value=0., input_shape=(self.Xw.shape[1], self.Xw.shape[2])))
model.add(GRU(units=units, return_sequences=True))
model.add(Dropout(0.2))
model.add(GRU(units=100, return_sequences=True))
model.add(Dropout(0.2))
model.add(GRU(units=units))
model.add(Dropout(0.2))
model.add(Dense(units=forecast)) 
model.compile(optimizer='adam', loss='mean_squared_error', metrics=['accuracy'])

**Comment by meliksahturker** [comment]
I confirm this.

**Comment by tinyxuyan** [comment]
I met the same issue using tensorflow 2.4 with python 3.6 on Ubuntu 18.04. Has the issue been solved? @CSharpDummy @meliksahturker

**Comment by sushreebarsa** [comment]
@tinyxuyan Could you please open a new ticket with all the relevant information regarding the issue that you are facing?
Thank you!
